### PR TITLE
README fixups

### PR DESCRIPTION
**Pull request by dokterbob**
Unpretentious README-basics. 

Just me trying to understand this project, it's dependencies and role within/relationship to DjangoFlow.

Unapologetically generated, arguably better than '...' as reported in #24.

I've currently added disclaimers:

> **_NOTE:_**  Generated content, as with human generated content, may contain errors or oversights. Please review before merging!

Happy to remove them, e.g. when project owner/maintainer judges instructions to be correct.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,85 +1,232 @@
...
 
+## Requirements
 
+> **_NOTE:_**  Generated content, as with human generated content, may contain errors or oversights. Please review before merging!
 
+Before integrating `django-df-chat` into your Django project, ensure that the following prerequisites are met:
 
+1. [Python](https://www.python.org/): This project requires Python 3.8 or newer.
+2. [Django](https://www.djangoproject.com/): Django version 3.2 or newer is required.
+3. [DjangoFlow](https://github.com/djangoflow): Django Flow, our base project with integrated CI/CD, is a recommended starting point for incorporating `django-df-chat`.
```
Comment (eugapx) [inline README.md:13]: This package can work without base djangoflow project, so I think we can remove this from dependencies

File: README.md [hunk 2]
```diff
@@ -1,85 +1,232 @@
...
+4. [Django REST Framework](https://www.django-rest-framework.org/): This project makes use of Django REST Framework for building APIs.
+5. [Django Channels](https://channels.readthedocs.io/en/latest/): This is needed for handling WebSockets and facilitating real-time communication.
+6. [Django Channels REST Framework](https://github.com/hishnash/djangochannelsrestframework): A library for building WebSocket API.
```
Comment (eugapx) [inline README.md:16]: Link to package was updated https://github.com/NilCoalescing/djangochannelsrestframework

File: README.md [hunk 3]
```diff
@@ -1,85 +1,232 @@
...
+7. [DRF Spectacular](https://drf-spectacular.readthedocs.io/en/latest/): A Django REST framework schema generation tool.
+8. [Django Hashid Field](https://github.com/nshafer/django-hashid-field): A reusable Django field that hides the fact that you're using AutoField or IntegerField primary keys in your objects.
+9. [DRF Nested Routers](https://github.com/alanjds/drf-nested-routers): This is used for managing CRUD operations on nested routes.
+10. [Django REST Framework SimpleJWT](https://django-rest-framework-simplejwt.readthedocs.io/en/latest/): This is used for authentication using JSON Web Tokens.
+11. [PyJWT](https://pyjwt.readthedocs.io/en/latest/): A Python library which allows you to encode and decode JSON Web Tokens (JWT).
+12. [Django Model Utils](https://django-model-utils.readthedocs.io/en/latest/): This project uses Django Model Utils for model mixins and utilities.
+13. [Django DF Notifications](https://github.com/dfurtado/django-notifications): This is used for managing user notifications in Django.
```
Comment (eugapx) [inline README.md:23]: Wrong link to DF notifications lib https://github.com/djangoflow/django-df-notifications

**Review by ayushin — COMMENTED** [review]
I believe that is is a dead end - we rather need to have an automated workflow to build the docs based on the actual code, or I would prefer to keep the all documentation in-line, there are tools for visualising it nicely.

**Comment by dokterbob** [comment]
@ayushin Nice to meet you!

These docs are in fact fully generated based on the actual code. As much as I'm with you on using something like pydoc for extracting class and method-specific documentation, I do consider it essential to have a small level of overview in the documentation, in addition to docstrings.

Specifically, for outsiders coming in (e.g. potential contributors, new team members), having a single point of access to get started with a project without having to read all the code and then tying it together saves a lot of time, motivates people and, last but not least, offloads leads/maintainers. It's effectively a FAQ to be more DRY and focused on moving projects forward if your usual questions are already answered and do not require asking team members the same question.

Especially outside users or contributors will be put off by this and simply not use an app which is not forthcoming in terms of documentation.

As it stands, I do think the README (largely based on the pre-existing structure, e.g. sections consisting solely of '...') is arguably a bit too verbose. Perhaps a short overview in the README for outsiders to judge whether this is the project for them and how to give it a minimal test-run, together with a link to ReadTheDocs is a good practice. The RTD could go into the how's and why's of architectural decisions and could include auto-generated class and method-level documentation.

**Pull request closed without merging**